native/spark-expr/src/scalar_funcs.rs

```diff
@@ -410,29 +410,31 @@ fn spark_rpad_internal<T: OffsetSizeTrait>(
     length: i32,
 ) -> Result<ColumnarValue, DataFusionError> {
     let string_array = as_generic_string_array::<T>(array)?;
+    let length = 0.max(length) as usize;
+    let space_string = " ".repeat(length);
+
+    let mut builder =
+        GenericStringBuilder::<T>::with_capacity(string_array.len(), string_array.len() * length);
 
-    let result = string_array
-        .iter()
-        .map(|string| match string {
+    for string in string_array.iter() {
+        match string {
             Some(string) => {
-                let length = if length < 0 { 0 } else { length as usize };
-                if length == 0 {
-                    Ok(Some("".to_string()))
+                // It looks Spark's UTF8String is closer to chars rather than graphemes
+                // https://stackoverflow.com/a/46290728
+                let char_len = string.chars().count();
+                if length <= char_len {
+                    builder.append_value(string);
```

Comment:
If the required len is less than string's length, don't we need to take substring of it? Spark RPad does it.

Reply:
Current implementation already has this issue.

Reply:
In the line 389 there is an existing comment
```
/// Similar to DataFusion `rpad`, but not to truncate when the string is already longer than length
```
Perhaps I should change the name of this method, this is not used for rpad